src/Appwrite/Platform/Modules/Proxy/Http/Rules/API/Create.php

```diff
@@ -0,0 +1,152 @@
+namespace Appwrite\Platform\Modules\Proxy\Http\Rules\API;
+
+use Appwrite\Event\Certificate;
+use Appwrite\Event\Event;
+use Appwrite\Extend\Exception;
+use Appwrite\Network\Validator\CNAME;
+use Appwrite\SDK\AuthType;
+use Appwrite\SDK\Method;
+use Appwrite\SDK\Response as SDKResponse;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate;
+use Utopia\Database\Helpers\ID;
+use Utopia\Domains\Domain;
+use Utopia\Platform\Action;
+use Utopia\Platform\Scope\HTTP;
+use Utopia\System\System;
+use Utopia\Validator\Domain as ValidatorDomain;
+
+class Create extends Action
+{
+    use HTTP;
+
+    public static function getName()
+    {
+        return 'createAPIRule';
+    }
+
+    public function __construct()
+    {
+        $this
+            ->setHttpMethod(Action::HTTP_REQUEST_METHOD_POST)
+            ->setHttpPath('/v1/proxy/rules/api')
+            ->groups(['api', 'proxy'])
+            ->desc('Create API rule')
+            ->label('scope', 'rules.write')
+            ->label('event', 'rules.[ruleId].create')
+            ->label('audits.event', 'rule.create')
+            ->label('audits.resource', 'rule/{response.$id}')
+            ->label('sdk', new Method(
+                namespace: 'proxy',
+                name: 'createAPIRule',
+                description: <<<EOT
+                Create a new proxy rule for serving Appwrite's API on custom domain.
+                EOT,
+                auth: [AuthType::ADMIN],
+                responses: [
+                    new SDKResponse(
+                        code: Response::STATUS_CODE_CREATED,
+                        model: Response::MODEL_PROXY_RULE,
+                    )
+                ]
+            ))
+            ->label('abuse-limit', 10)
+            ->label('abuse-key', 'userId:{userId}, url:{url}')
+            ->label('abuse-time', 60)
+            ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+            ->inject('response')
+            ->inject('project')
+            ->inject('queueForCertificates')
+            ->inject('queueForEvents')
+            ->inject('dbForPlatform')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        $deniedDomains = [
+            $mainDomain,
+            $sitesDomain,
+            $functionsDomain,
+            'localhost',
+            APP_HOSTNAME_INTERNAL,
+        ];
+
+        if (\in_array($domain, $deniedDomains)) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please pick another one.');
```

Comment:
Should we specify here that there could be a type mismatch. Like if you passed `type` as API by mistake along with sites or functions domain.

Reply:
We no longer accept `type` parameter. We have 3 separate methods / endpoints

Reply:
Makes sense